core/app/models/spree/reimbursement.rb

```diff
@@ -59,6 +59,13 @@ class IncompleteReimbursementError < StandardError; end
     class_attribute :reimbursement_failure_hooks
     self.reimbursement_failure_hooks = []
 
+    # The legacy_store_credit_category_name allows to control whether the legacy
+    # way of fetching the category should be used.
+    #
+    # @param [Boolean] enable/disable the legacy way of fetching the store category name
+    class_attribute :legacy_store_credit_category_name
+    self.legacy_store_credit_category_name = false
```

Comment:
we should start with this preference set true, so people using it has the chance to be notified that we are going to switch the value in Solidus 3.0.

Reply:
I finally moved it to an app config, that way support for testing and otherwise was much better